core/src/main/java/com/sequenceiq/cloudbreak/service/events/DefaultCloudbreakEventService.java

```diff
@@ -108,6 +121,19 @@ public void fireCloudbreakInstanceGroupEvent(Long stackId, String eventType, Str
         fireEventWithPayload(stackId, eventType, resourceEvent, eventMessageArgs, eventMessage, instanceGroupName);
     }
 
+    @Override
+    public void fireClusterManagerEvent(Stack stack, ResourceEvent resourceEvent, String eventName, Optional<BigDecimal> clusterManagerEventId) {
+        if (clusterManagerEventId.isPresent()) {
+            Optional<String> link = cmCommandLinkProvider.getCmCommandLink(stack, clusterManagerEventId.get().toString());
+            if (link.isPresent()) {
```

Comment:
Is it intentional not to send event if no link is provided? The _Cloudera Manager command failed ..._ part of error message seems still valid for me in this case.

Reply:
Yes, this is intentional. Unfortunately I had to send an extra notification with the link, there is always a notification about the failure, but at that time the command id is not present, so I had to add the logic to send an extra event with the link, so without the link the extra notification isn't needed.

Reply:
Unfortunately there are various error handling logics implemented for exceptions from CM command pollers in various places (flow, service layer, etc.), so there is no easy way to add the link to the existing events, as far as I have seen it in the code.